actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -406,6 +407,16 @@ def ssl?
       super || scheme == 'wss'.freeze
     end
 
+    def method_missing(method, *args)
+      return unless Rack::Request.public_instance_methods.include?(method)
+      ActiveSupport::Deprecation.warn(<<-MESSAGE.strip_heredoc)
+              ActionDispatch::Request will no longer inherit from Rack::Request in Rails 5.1.
+              The method '#{method}' and all other public API exposed by ActionDispatch::Request via its inheritance of Rack::Request will be removed without replacement.
+            MESSAGE
+
+      Rack::Request.new(env).send(method, *args)
```

Comment:
This is a question for you to answer :grin: – Audit the AD::Request code to verify that it isn't keeping local state (e.g. memoized ivars) that would cause surprising behavior when hitting Rack::Request (like, reparsing cookies, or parsing params differently, or …).